From f779158421fbf746d101568f86539dfb05c836d6 Mon Sep 17 00:00:00 2001
From: Christophe Chapuis <chris.chapuis@gmail.com>
Date: Sat, 5 May 2018 13:41:35 +0000
Subject: [PATCH] Fix build with Wayland SDL2

* this was partially resolved in:
    commit 1a5c19ec51e76ddaec469d1231e6bdb087323d79
    Author: Simon Fels <morphis@gravedo.de>
    Date:   Mon Jul 23 16:37:54 2018 +0200

    sdl: take native window and display for wayland from SDL
  but in our build with x11 support disabled it was still failing with:
  anbox/3.0+gitAUTOINC+d521e28296-r0/git/src/anbox/platform/sdl/window.cpp:84:69: error: 'union SDL_SysWMinfo::<unnamed>' has no member named 'x11'
  disable it with X11_SUPPORT define (which currently isn't set
  in configure unlike e.g. WAYLAND_SUPPORT, so it's always disabled
  for now)

Signed-off-by: Christophe Chapuis <chris.chapuis@gmail.com>
Signed-off-by: Martin Jansa <Martin.Jansa@gmail.com>
---
 src/anbox/platform/sdl/window.cpp | 2 ++
 src/anbox/ui/splash_screen.cpp    | 2 +-
 2 files changed, 3 insertions(+), 1 deletion(-)

diff --git a/src/anbox/platform/sdl/window.cpp b/src/anbox/platform/sdl/window.cpp
index 752b13c..13d059e 100644
--- a/src/anbox/platform/sdl/window.cpp
+++ b/src/anbox/platform/sdl/window.cpp
@@ -80,10 +80,12 @@ Window::Window(const std::shared_ptr<Renderer> &renderer,
   SDL_VERSION(&info.version);
   SDL_GetWindowWMInfo(window_, &info);
   switch (info.subsystem) {
+#if defined(X11_SUPPORT)
     case SDL_SYSWM_X11:
       native_display_ = static_cast<EGLNativeDisplayType>(info.info.x11.display);
       native_window_ = static_cast<EGLNativeWindowType>(info.info.x11.window);
       break;
+#endif
 #if defined(WAYLAND_SUPPORT)
     case SDL_SYSWM_WAYLAND:
       native_display_ = static_cast<EGLNativeDisplayType>(info.info.wl.display);
diff --git a/src/anbox/ui/splash_screen.cpp b/src/anbox/ui/splash_screen.cpp
index 4afdaf0..299be03 100644
--- a/src/anbox/ui/splash_screen.cpp
+++ b/src/anbox/ui/splash_screen.cpp
@@ -20,7 +20,7 @@
 #include "anbox/utils.h"
 #include "anbox/logger.h"
 
-#include <SDL2/SDL_image.h>
+#include <SDL_image.h>
 
 namespace anbox {
 namespace ui {
